packages/commonwealth/server/controllers/server_communities_methods/create_community.ts

```diff
@@ -257,6 +222,40 @@ export async function __createCommunity(
     community.base === ChainBase.CosmosSDK &&
     community.type !== ChainType.Offchain
   ) {
+    // cosmos_chain_id is the canonical identifier for a cosmos chain.
+    // Our convention is to follow the "chain_name" standard established by the
+    // Cosmos Chain Registry:
+    // https://github.com/cosmos/chain-registry/blob/dbec1643b587469383635fd345634fb19075b53a/chain.schema.json#L1-L20
+    // This community-led registry seeks to track chain info for all Cosmos chains.
+    // The primary key for a chain there is "chain_name." This is our cosmos_chain_id.
+    // It is a lowercase alphanumeric name, like 'osmosis'.
+    // See: https://github.com/hicommonwealth/commonwealth/issues/4951
+    cosmos_chain_id = community.cosmos_chain_id || null;
+
+    if (!cosmos_chain_id) {
+      throw new AppError(Errors.CosmosChainNameRequired);
+    } else {
+      const oldChainNode = await this.models.ChainNode.findOne({
+        where: { cosmos_chain_id },
+      });
+      if (oldChainNode && oldChainNode.cosmos_chain_id === cosmos_chain_id) {
+        throw new AppError(`${Errors.ChainNodeIdExists}: ${cosmos_chain_id}`);
+      }
+    }
+
+    const REGISTRY_API_URL = 'https://cosmoschains.thesilverfox.pro';
```

Comment:
Should this be an environment variable declared in config.ts?

Reply:
I think so, it should stay as an environment variable but leaving it as it is will not cause any specific problem while we do not use any specific auth key or user credentials to send a request to that API endpoint.

Reply:
Agree. It would be good to be able to use an alternative API if this one fails. Env var now.